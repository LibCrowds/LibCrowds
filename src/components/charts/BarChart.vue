<template>
  <b-card :header="header">
    <div class="ct-container" ref="chart"></div>
  </b-card>
</template>

<script>
import Chartist from 'chartist'
import 'chartist-plugin-tooltips'
import pluralize from 'pluralize'

export default {
  data: function () {
    return {
      opts: {
        height: '300px',
        plugins: [
          Chartist.plugins.tooltip({
            transformTooltipTextFnc: (val) => {
              let text = pluralize(this.unit, Number(val))
              return `${val} ${text}`
            }
          })
        ]
      }
    }
  },

  props: {
    data: {
      type: Object,
      required: true
    },
    unit: {
      type: String,
      required: true
    },
    header: {
      type: String,
      required: true
    }
  },

  mounted () {
    const chart = this.$refs.chart
    Chartist.Bar(chart, this.data, this.opts)
  }
}
</script>
