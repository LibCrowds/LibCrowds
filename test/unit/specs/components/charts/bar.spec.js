import BootstrapVue from 'bootstrap-vue'
import VueChartist from 'vue-chartist'
import 'chartist-plugin-tooltips'

import { mount, createLocalVue } from 'vue-test-utils'
import BarChart from '~/components/charts/Bar'

describe('Bar chart', () => {
  let localVue = null
  let wrapper = null

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(BootstrapVue)
    localVue.use(VueChartist)
    wrapper = mount(BarChart, {
      localVue,
      propsData: {
        header: 'My Bar Chart',
        chartData: {
          labels: ['A', 'B', 'C'],
          series: [[1, 3, 2], [4, 6, 5]]
        },
        unit: 'task'
      }
    })
  })

  it('renders correctly', () => {
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })
})
