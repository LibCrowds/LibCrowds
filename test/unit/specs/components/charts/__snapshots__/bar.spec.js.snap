// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar chart renders correctly 1`] = `

<div data-server-rendered="true"
     class="card"
>
  <div class="card-header">
    <div>
      My Bar Chart
    </div>
  </div>
  <div class="card-body">
    <div>
    </div>
  </div>
</div>

`;
